clouddriver-kubernetes/src/integration/java/com/netflix/spinnaker/clouddriver/kubernetes/it/DeleteManifestIT.java

```diff
@@ -0,0 +1,157 @@
+        .withValue("deleteManifest.location", account1Ns)
+        .withValue("deleteManifest.account", ACCOUNT1_NAME)
+        .asList();
+  }
+
+  public List<Map<String, Object>> buildDynamicRequestBody(
+      String manifestName, Boolean cascading, String cluster, String criteria, String kind) {
+    return KubeTestUtils.loadJson("classpath:requests/delete_manifest.json")
+        .withValue("deleteManifest.app", APP_1_NAME)
+        .withValue("deleteManifest.mode", "dynamic")
+        .withValue("deleteManifest.cluster", cluster)
+        .withValue("deleteManifest.criteria", criteria)
+        .withValue("deleteManifest.kind", kind)
+        .withValue("deleteManifest.manifestName", manifestName)
+        .withValue("deleteManifest.options.cascading", cascading)
+        .withValue("deleteManifest.location", account1Ns)
+        .withValue("deleteManifest.account", ACCOUNT1_NAME)
+        .asList();
+  }
+
+  public void deploy(final List<Map<String, Object>> manifest, String... objectNames)
+      throws InterruptedException {
+    List<Map<String, Object>> b =
+        KubeTestUtils.loadJson("classpath:requests/deploy_manifest.json")
+            .withValue("deployManifest.account", ACCOUNT1_NAME)
+            .withValue("deployManifest.namespaceOverride", account1Ns)
+            .withValue("deployManifest.moniker.app", APP_1_NAME)
+            .withValue("deployManifest.manifests", manifest)
+            .asList();
+    KubeTestUtils.deployAndWaitStable(baseUrl(), b, account1Ns, objectNames);
+  }
+
+  public void deployReplicaSets(byte amount) throws InterruptedException {
+    List<Map<String, Object>> m =
+        KubeTestUtils.loadYaml("classpath:manifests/replicaset.yml").asList();
+    for (byte i = 0; i < amount; i++) {
+      deploy(m, String.format("replicaSet nginx-v%03d", i));
+    }
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given an existing static target\n"
+          + "When sending delete manifest request\n"
+          + "It should wait until the target is deleted\n===")
+  @Test
+  public void staticTarget() throws IOException, InterruptedException {
+    final String sn = "mysecret";
+    Map<String, Object> sm =
+        KubeTestUtils.loadYaml("classpath:manifests/secret.yml")
+            .withValue("metadata.namespace", account1Ns)
+            .withValue("metadata.name", sn)
+            .asMap();
+    kubeCluster.execKubectl("-n " + account1Ns + " apply -f -", sm);
+    List<Map<String, Object>> rb = buildStaticRequestBody(String.format("secret %s", sn), true);
+    List<String> on = KubeTestUtils.delete(baseUrl(), rb, account1Ns);
+    assertEquals(1, on.size());
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given a existing dynamic target\n"
+          + "When sending delete manifest request\n"
+          + "It should wait until the target is deleted\n===")
+  @Test
+  public void dynamicTarget() throws InterruptedException, IOException {
+    deployReplicaSets((byte) 2);
```

Comment:
It's preferable to use here the `KubeTestUtils.deployAndWaitStable` method instead of helper methods, even if that leads to duplicate code in test cases. The reason is that it's easier to understand and fix a failed test case if we know exactly what's doing without digging in too much nested helper methods.